packages/shared/src/hooks/useFeedSettings.ts

```diff
@@ -24,25 +33,64 @@ export type FeedSettingsReturnType = {
 
 let avoidRefresh = false;
 
+export const LOCAL_FEED_SETTINGS_KEY = 'feedSettings:local';
+
+export const updateLocalFeedSettings = (feedSettings: FeedSettings): void => {
+  storage.setItem(LOCAL_FEED_SETTINGS_KEY, JSON.stringify(feedSettings));
+};
+
+export const getLocalFeedSettings = (): FeedSettings => {
+  const value = storage.getItem(LOCAL_FEED_SETTINGS_KEY);
+  if (!value) {
+    return getEmptyFeedSettings();
+  }
+
+  try {
+    return JSON.parse(value) as FeedSettings;
+  } catch (ex) {
+    return getEmptyFeedSettings();
+  }
+};
+
+const isObjectEmpty = (obj: unknown) => {
+  if (typeof obj === 'undefined' || obj === null) {
+    return true;
+  }
+
+  return Object.keys(obj).length === 0;
+};
+
 export default function useFeedSettings(): FeedSettingsReturnType {
-  const { user } = useContext(AuthContext);
+  const { user, loadedUserFromCache } = useContext(AuthContext);
+  const client = useQueryClient();
   const [isFirstLoad, setIsFirstLoad] = useState(true);
   const filtersKey = getFeedSettingsQueryKey(user);
   const queryClient = useQueryClient();
+  const { flags } = useContext(FeaturesContext);
+  const myFeed = getFeatureValue(Features.MyFeedOn, flags, 'false');
+  const shouldShowMyFeed = myFeed === 'true';
   const setAvoidRefresh = (value: boolean) => {
     avoidRefresh = value;
   };
 
-  const {
-    data: { tagsCategories, feedSettings, advancedSettings } = {},
-    isLoading,
-  } = useQuery<AllTagCategoriesData>(filtersKey, () =>
-    request(`${apiUrl}/graphql`, FEED_SETTINGS_QUERY, {
-      loggedIn: !!user?.id,
-    }),
+  const { data: feedQuery = {}, isLoading } = useQuery<AllTagCategoriesData>(
+    filtersKey,
+    async () => {
+      const req = await request(`${apiUrl}/graphql`, FEED_SETTINGS_QUERY, {
+        loggedIn: !!user?.id,
+      });
+
+      return { ...feedQuery, ...req };
+    },
   );
 
+  const { tagsCategories, feedSettings, advancedSettings } = feedQuery;
+
   useEffect(() => {
+    if (!user?.id) {
+      return;
+    }
+
     if (avoidRefresh) {
       return;
     }
```

Comment:
I'm guessing we can combine these two right?

Reply:
Ah, totally makes sense